service/history/historyCache_test.go

```diff
@@ -76,37 +88,72 @@ func (s *historyCacheSuite) SetupTest() {
 	s.cache = newHistoryCache(s.mockShard, s.logger)
 }
 
+func (s *historyCacheSuite) TearDownTest() {
+	s.mockExecutionMgr.AssertExpectations(s.T())
+}
+
 func (s *historyCacheSuite) TestHistoryCachePinning() {
 	s.mockShard.GetConfig().HistoryCacheMaxSize = 2
-	domain := "test_domain"
+	domainID := "test_domain_id"
 	s.cache = newHistoryCache(s.mockShard, s.logger)
 	we := workflow.WorkflowExecution{
-		WorkflowId: common.StringPtr("wf-cache-test"),
+		WorkflowId: common.StringPtr("wf-cache-test-pinning"),
 		RunId:      common.StringPtr(uuid.New()),
 	}
 
-	context, release, err := s.cache.getOrCreateWorkflowExecution(domain, we)
+	context, release, err := s.cache.getOrCreateWorkflowExecution(domainID, we)
 	s.Nil(err)
 
 	we2 := workflow.WorkflowExecution{
-		WorkflowId: common.StringPtr("wf-cache-test"),
+		WorkflowId: common.StringPtr("wf-cache-test-pinning"),
 		RunId:      common.StringPtr(uuid.New()),
 	}
 
 	// Cache is full because context is pinned, should get an error now
-	_, _, err2 := s.cache.getOrCreateWorkflowExecution(domain, we2)
+	_, _, err2 := s.cache.getOrCreateWorkflowExecution(domainID, we2)
 	s.NotNil(err2)
 
 	// Now release the context, this should unpin it.
-	release()
+	release(err2)
 
-	_, release2, err3 := s.cache.getOrCreateWorkflowExecution(domain, we2)
+	_, release2, err3 := s.cache.getOrCreateWorkflowExecution(domainID, we2)
 	s.Nil(err3)
-	release2()
+	release2(err3)
 
 	// Old context should be evicted.
-	newContext, release, err4 := s.cache.getOrCreateWorkflowExecution(domain, we)
+	newContext, release, err4 := s.cache.getOrCreateWorkflowExecution(domainID, we)
 	s.Nil(err4)
 	s.False(context == newContext)
-	release()
+	release(err4)
+}
+
+func (s *historyCacheSuite) TestHistoryCacheClear() {
```

Comment:
[NOT A BLOCKER for this CR ] can you also add a test that exercises mutable state reload code path (if it doesn't exist already ?) i.e. cache hit followed by clear() due to error followed by mutable state reload

Reply:
done in the update